compiler/const-manipulations.h

```diff
@@ -246,13 +247,20 @@ struct CheckConstAccess final
 };
 
 // moved from the collect-required-and-classes.cpp to make it more reusable
-static inline std::string collect_string_concatenation(VertexPtr v) {
+static inline std::string collect_string_concatenation(VertexPtr v, bool allow_dirname = false) {
   if (auto string = v.try_as<op_string>()) {
     return string->str_val;
   }
+  if (auto call = v.try_as<op_func_call>(); allow_dirname && call && call->str_val == "dirname") {
+    kphp_assert_msg(call->size() == 1, "call to 'dirname' must have one arg");
```

Comment:
I think it makes sense to use a "compile error" level here and add a `kphp_should_fail` test that verifies that we give an expected error.
`kphp_should_fail` are cheap and executed almost instantly.

Reply:
done